pkg/asset/agent/image/agentpxefiles.go

```diff
@@ -169,3 +184,25 @@ func (a *AgentPXEFiles) copy(filepath string, src io.Reader) error {
 
 	return nil
 }
+
+func (a *AgentPXEFiles) createiPXEScript(cpuArch, ipxeBaseURL, pxeAssetsFullPath string) error {
+	iPXEScriptTemplate := `#!ipxe 
+initrd --name initrd %s/%s 
+kernel %s/%s  initrd=initrd coreos.live.rootfs_url=%s/%sg random.trust_cpu=on rd.luks.options=discard ignition.firstboot ignition.platform.id=metal 
```

Comment:
Where are coming from the kargs `random.trust_cpu=on rd.luks.options=discard`?

Reply:
We can remove it. They were originally referred from [here](https://github.com/openshift/assisted-service/blob/b633a3ab4b9207aeff323cafcd748442e551d39d/docs/user-guide/deploy-on-bare-metal.md?plain=1#L55)

Reply:
If not an issue, I'd keep the bare minimum amount of strictly required kargs in the config file

Reply:
We should keep the same kargs as we use in the ISO (which don't include these two), modulo the ones that are pxe-specific, so that we minimise behavioural differences.
It's possible that those kargs are added even to the ISO in the assisted discovery image, and if that's the case maybe we should consider adding them (in both places) too. But that isn't part of this work.

Reply:
Also the `g` at the end of `coreos.live.rootfs_url=%s/%sg` looks suspect.